### PR TITLE
v18 comparative RNA-seq fixes

### PR DESCRIPTION
**Pull request by cansavvy**
#### What scientific question is your analysis addressing?

v18 brought about some errors in the Comparative RNA-seq runs. This PR is a work in progress toward investigating and resolving those. 

#### What was your approach?

So far: 
- As #890 notes, there's some extraneous files in `pbta-mend-qc-results.tar.gz`. To get around this, 01-correlation-matrix.py has been changed to only read in the files that are listed in the `pbta-mend-qc-manifest.tsv` file. 

An error still at large, `prepare_expression` encounters a line of NAs in one of the files. 
```
prepare_expression
    expression_filtered_df = samples[has_few_enough_zeroes]
  File "/usr/local/lib/python3.5/dist-packages/pandas/core/frame.py", line 3001, in __getitem__
    indexer = self.loc._convert_to_indexer(key, axis=1, raise_missing=True)
  File "/usr/local/lib/python3.5/dist-packages/pandas/core/indexing.py", line 1285, in _convert_to_indexer
    return self._get_listlike_indexer(obj, axis, **kwargs)[1]
  File "/usr/local/lib/python3.5/dist-packages/pandas/core/indexing.py", line 1092, in _get_listlike_indexer
    keyarr, indexer, o._get_axis_number(axis), raise_missing=raise_missing
  File "/usr/local/lib/python3.5/dist-packages/pandas/core/indexing.py", line 1177, in _validate_read_indexer
    key=key, axis=self.obj._get_axis_name(axis)
KeyError: "None of [Float64Index([nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,\n              ...\n              nan, nan, nan, nan, nan, nan, nan, nan, nan, nan],\n             dtype='float64', length=58347)] are in the [columns]"
```

#### What GitHub issue does your pull request address?

Related to #883 and #890 

#### Reproducibility Checklist

- [ ] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [ ] This analysis has been added to continuous integration.

#### Documentation Checklist

- [ ] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [ ] The analytical code is documented and contains comments.

### CODE REVIEW

**Comment by jaclyn-taroni** [comment]
I think you meant to request a review from @jashapiro rather than assign.

**Comment by jashapiro** [comment]
> I think you meant to request a review from @jashapiro rather than assign.

Nope, assign is correct! We fixed one problem but there is more to do yet!

**Comment by jashapiro** [comment]
This is now working, but I would prefer to revert 4373608d166907121f26f26a25175dca5b3c3d67 and df2511c94d2544819f234b38bc999e0f58a74622 once #890 is resolved, as I have not done extensive testing. Note also that this should be rerun with v18 to generate up to date results including new samples after v18 is merged.

**Comment by jashapiro** [comment]
The fix of #890 presumably would make this work with the original code, but this PR still includes some code changes to make things a bit more robust. Notably, it also changes the way the code is run in CI and later, by now making the python scripts run "normally" with file arguments specified relative the location from which the code is run, not the location of the script itself. The new shell script that runs the full workflow _does_ run from its location, but takes no arguments itself.

File: analyses/comparative-RNASeq-analysis/run-comparative-RNAseq.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+# CCDL for ALSF 2021
+# Joshua Shapiro
+#
+# Set so the script stops if there is an error
+set -euo pipefail
+
+# Set the working directory to the directory of this file
+cd "$(dirname "${BASH_SOURCE[0]}")"
+
```
Comment (jaclyn-taroni) [inline analyses/comparative-RNASeq-analysis/run-comparative-RNAseq.sh:10]: Very minor comment: might want to use a variable for `../../data` here.

**Review by jaclyn-taroni — APPROVED** [review]
Looks good to me. I did not run this locally, but CI passes so I assume that the paths are all a-okay! I had one minor comment.

File: analyses/comparative-RNASeq-analysis/run-comparative-RNAseq.sh (hunk 1 above)
Comment (jashapiro) [reply analyses/comparative-RNASeq-analysis/run-comparative-RNAseq.sh:10]: done!

**Pull request merged**